Incompatibility of lmdb-master-sys Package with Cross-Compilation

#### Description

The `lmdb-master-sys` package in the `heed` crate does not support cross-compilation correctly. When attempting to cross-compile using the command:

```sh
cargo build --target aarch64-unknown-linux-gnu
```

the library generates incompatible binary files:

```sh
8c28d618c08bdb0c-mdb.o:                                                         ELF 64-bit LSB relocatable, x86-64, version 1 (SYSV), with debug_info, not stripped
8c28d618c08bdb0c-midl.o:                                                        ELF 64-bit LSB relocatable, x86-64, version 1 (SYSV), with debug_info, not stripped
lmdb_master_sys-f5d577e22e9dc6e8.lmdb_master_sys.4026053b71a9477b-cgu.0.rcgu.o: ELF 64-bit LSB relocatable, ARM aarch64, version 1 (SYSV), not stripped
```

This results in a linking error:

```sh
aarch64-unknown-linux-gnu-ld: release/deps/liblmdb_master_sys-4cb3ac6e86087a04.rlib: error adding symbols: file in wrong format
collect2: error: ld returned 1 exit status
```

#### Context

Some libraries, such as `zstd`, can resolve cross-compilation issues by setting environment variables:

```sh
export PKG_CONFIG_ALLOW_CROSS=1
export ZSTD_SYS_USE_PKG_CONFIG=1
```

However, it appears that `lmdb-master-sys` does not utilize `pkg-config` for its configuration. This results in the generation of object files for the host architecture instead of the target architecture.

#### Steps to Reproduce

1. Set up a cross-compilation environment for `aarch64-unknown-linux-gnu`.
2. Attempt to build a project that includes the `heed` crate with the following command:

```sh
cargo build --target aarch64-unknown-linux-gnu
```

3. Observe the generated object files and the resulting linking error.

#### Expected Behavior

The `lmdb-master-sys` package should correctly generate object files for the target architecture (`aarch64-unknown-linux-gnu`) and link them without errors.

#### Actual Behavior

The package generates object files for the host architecture (`x86-64`), leading to a linking error.

#### Additional Information

- `cargo` version: cargo 1.80.1 (376290515 2024-07-16)
- `rustc` version: rustc 1.80.1 (3f5fd8dd4 2024-08-06)
- Target: `aarch64-unknown-linux-gnu`
- Host: `x86_64-unknown-linux-gnu`

Thank you for your attention to this issue. Please let me know if you need any further information or assistance in reproducing the problem.

--- a/heed-types/Cargo.toml
+++ b/heed-types/Cargo.toml
@@ -12,8 +12,8 @@ edition = "2021"
 bincode = { version = "1.3.3", optional = true }
 byteorder = "1.5.0"
 heed-traits = { version = "0.20.0", path = "../heed-traits" }
-serde = { version = "1.0.217", optional = true }
-serde_json = { version = "1.0.135", optional = true }
+serde = { version = "1.0.218", optional = true }
+serde_json = { version = "1.0.140", optional = true }
 rmp-serde = { version = "1.3.0", optional = true }
 
 [features]

--- a/heed/Cargo.toml
+++ b/heed/Cargo.toml
@@ -20,21 +20,21 @@ include = [
 ]
 
 [dependencies]
-bitflags = { version = "2.6.0", features = ["serde"] }
+bitflags = { version = "2.9.0", features = ["serde"] }
 byteorder = { version = "1.5.0", default-features = false }
 heed-traits = { version = "0.20.0", path = "../heed-traits" }
 heed-types = { version = "0.21.0", default-features = false, path = "../heed-types" }
-libc = "0.2.169"
+libc = "0.2.170"
 lmdb-master-sys = { version = "0.2.4", path = "../lmdb-master-sys" }
-once_cell = "1.20.2"
+once_cell = "1.20.3"
 page_size = "0.6.0"
-serde = { version = "1.0.217", features = ["derive"], optional = true }
+serde = { version = "1.0.218", features = ["derive"], optional = true }
 synchronoise = "1.0.1"
 
 [dev-dependencies]
 memchr = "2.7.4"
-serde = { version = "1.0.217", features = ["derive"] }
-tempfile = "3.15.0"
+serde = { version = "1.0.218", features = ["derive"] }
+tempfile = "3.18.0"
 
 [target.'cfg(windows)'.dependencies]
 url = "2.5.4"

--- a/lmdb-master-sys/Cargo.toml
+++ b/lmdb-master-sys/Cargo.toml
@@ -24,13 +24,13 @@ name = "lmdb_master_sys"
 doctest = false
 
 [dependencies]
-libc = "0.2.169"
+libc = "0.2.170"
 
 [build-dependencies]
 bindgen = { version = "0.71.1", default-features = false, optional = true, features = [
     "runtime",
 ] }
-cc = "1.2.7"
+cc = "1.2.16"
 doxygen-rs = "0.4.2"
 
 [dev-dependencies]

--- a/lmdb-master3-sys/Cargo.toml
+++ b/lmdb-master3-sys/Cargo.toml
@@ -24,13 +24,13 @@ name = "lmdb_master3_sys"
 doctest = false
 
 [dependencies]
-libc = "0.2.169"
+libc = "0.2.170"
 
 [build-dependencies]
 bindgen = { version = "0.71.1", default-features = false, optional = true, features = [
     "runtime",
 ] }
-cc = "1.2.7"
+cc = "1.2.16"
 doxygen-rs = "0.4.2"
 
 [dev-dependencies]

